numpy/linalg/linalg.py

```diff
@@ -1138,11 +1140,12 @@ def eig(a):
 
     """
     a, wrap = _makearray(a)
-    _assertNoEmpty2d(a)
     _assertRankAtLeast2(a)
     _assertNdSquareness(a)
     _assertFinite(a)
     t, result_t = _commonType(a)
+    if product(a.shape[-2:]) == 0:
+        return empty(a.shape[-1:], dtype=result_t), wrap(empty(a.shape, dtype=result_t))
```

Comment:
I'd just split this into two lines for computing the returned object, then the return. It takes up a bit more space but shortens the line and is clearer (IMHO).

Same for other long lines.